src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -243,12 +251,48 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
                     }
 
                     // attempt restarts using an exponential backoff strategy
-                    CreateRestartBackoffDelay(consecutiveErrorCount).GetAwaiter().GetResult();
+                    CreateRestartBackoffDelay(_consecutiveErrorCount).GetAwaiter().GetResult();
                 }
             }
             while (!_stopped && !cancellationToken.IsCancellationRequested);
         }
 
+        private bool ShutdownHostIfUnhealthy()
+        {
+            if (ShouldMonitorHostHealth && _healthCheckWindow.GetEvents(isHealthy => !isHealthy).Count() > _config.HostHealthMonitor.HealthCheckThreshold)
+            {
+                // if the number of times the host has been unhealthy in
+                // the current time window exceeds the threshold, recover by
+                // initiating shutdown
+                var message = $"Host unhealthy count exceeds the threshold of {_config.HostHealthMonitor.HealthCheckThreshold} for time window {_config.HostHealthMonitor.HealthCheckWindow}. Initiating shutdown.";
+                _traceWriter?.Error(message);
+                _logger?.LogError(0, message);
+                Shutdown();
+                return true;
+            }
+
+            return false;
+        }
+
+        private void OnHostInitializing(object sender, EventArgs e)
+        {
+            // by the time this event is raised, loggers for the host
+            // have been initialized
+            var host = (ScriptHost)sender;
+            _traceWriter = host.TraceWriter;
+            _logger = host.Logger;
+
+            string extensionVersion = _settingsManager.GetSetting(EnvironmentSettingNames.FunctionsExtensionVersion);
+            string hostId = host.ScriptConfig.HostConfig.HostId;
+            string message = $"Starting Host (HostId={hostId}, Version={ScriptHost.Version}, ProcessId={Process.GetCurrentProcess().Id}, AppDomainId={AppDomain.CurrentDomain.Id}, Debug={host.InDebugMode}, ConsecutiveErrors={_consecutiveErrorCount}, StartupCount={_hostStartCount}, FunctionsExtensionVersion={extensionVersion})";
```

Comment:
In addition to moving this log statement from above, I've also added AppDomainId logging.